crowbar_framework/app/assets/javascripts/barclamps/neutron/application.js

```diff
@@ -253,26 +253,52 @@ function networking_plugin_check() {
 }
 
 function ml2_type_drivers_check() {
-  var values = $('#ml2_type_drivers').val();
+  var values = $('#ml2_type_drivers').val() || [];
   if (values.indexOf("vlan") >= 0) {
     $('#num_vlans_container').show();
   } else {
     $('#num_vlans_container').hide();
   }
+
+  if (values.length == 1) {
```

Comment:
> This logic doesn't work. When doing the following, I get The default provider network type driver "vlan" is not a selected ml2 type driver as error.

This is weird. I can see that behaviour as well. But it doesn't seem to be an issue with this change. At least I can see the values of the combo boxes correctly updated to only remaining selected type_driver. I'll investigate further.

> Another problem is, that the default ml2 type driver boxes are visible after pressing Apply/Save (and no error occurs) even if only one type driver is availalbe.

This is what @vuntz  already mentioned. Will be fixed with the next iteration.